### PR TITLE
Raise error for streamAgg when plan contains group by key

### LINKED ISSUE #6381: Raise error for streamAgg when plan contains group by key.
## Enhancement

related to https://github.com/pingcap/tidb/issues/39266

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #6381 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Planner/PhysicalPlan.cpp [hunk 1]
```diff
@@ -116,8 +118,9 @@ void PhysicalPlan::build(const String & executor_id, const tipb::Executor * exec
             pushBack(PhysicalFilter::build(context, executor_id, log, executor->selection(), child));
         break;
     }
-    case tipb::ExecType::TypeAggregation:
     case tipb::ExecType::TypeStreamAgg:
+        RUNTIME_CHECK_MSG(executor->aggregation().group_by_size() == 0, STREAM_AGG_ERROR);
```
Comment (SeaRise) [inline dbms/src/Flash/Planner/PhysicalPlan.cpp:122]: why not
```suggestion
        RUNTIME_CHECK_MSG(executor->aggregation().group_by_size() == 0, "Group by key is not supported in StreamAgg");
```

**Review by windtalker — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]

**Pull request merged**